### PR TITLE
G-API. Face detection MTCNN demo.

### PR DESCRIPTION
**Pull request by mpashchenkov**
The G-API version of OMZ Python face detection MTCNN demo.

### CODE REVIEW

File: demos/tests/cases.py [hunk 1]
```diff
@@ -158,6 +158,17 @@ def single_option_cases(key, *args):
         }),
     )),
 
+    CppDemo(name='face_detection_mtcnn_demo', implementation='cpp_gapi',
+            device_keys=['-d_p', '-d_o'],
```
Comment (eizamaliev) [inline demos/tests/cases.py:162]:
```suggestion
            device_keys=['-d_p', '-d_r', '-d_o'],
```

Comment (mpashchenkov) [reply demos/tests/cases.py]: Yes, thanks. Can mtcnn composite network work with different devices? OMZ demo has two cases: all on CPU and all on GPU.  Does different devices make sense?

Also OMZ demo has bad performance for case p-net and r-net on GPU.

Comment (eizamaliev) [reply demos/tests/cases.py]: The OMZ demo has any case, but for testing uses only one device for all models.

Comment (mpashchenkov) [reply demos/tests/cases.py]: I think it hasn't. [Only all CPU or all GPU](https://github.com/openvinotoolkit/open_model_zoo/blob/b15ae77b045eaa596c3ec453117bc37e3420a5d8/demos/face_detection_mtcnn_demo/python/face_detection_mtcnn_demo.py#L64)

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    pipeline_mtcnn.start();
+    while (pipeline_mtcnn.pull(cv::gout(out_image))) {
+        frames++;
+
+        tm.stop();
+        const auto fps_str = std::to_string(int(frames / tm.getTimeSec())) + " FPS";
+        cv::putText(out_image, fps_str, { 0,32 }, cv::FONT_HERSHEY_SIMPLEX, 1.0, { 0, 0, 255 }, 3);
+
+        if (!FLAGS_no_show) {
+            cv::waitKey(1);
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp:204]: Add exit by `ESC` or `Q` key.

**Review by eizamaliev — CHANGES_REQUESTED** [review]

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    cv::TickMeter tm, tm_all;
+    tm.start();
+    tm_all.start();
+    pipeline_mtcnn.start();
+    while (pipeline_mtcnn.pull(cv::gout(out_image))) {
+        frames++;
+
+        tm.stop();
+        const auto fps_str = std::to_string(int(frames / tm.getTimeSec())) + " FPS";
```
Comment (mpashchenkov) [inline demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp:200]: `.1f` instead static_cast<int>() for FPS counter.

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        cv::putText(out_image, fps_str, { 0,32 }, cv::FONT_HERSHEY_SIMPLEX, 1.0, { 0, 0, 255 }, 3);
```
Comment (mpashchenkov) [inline demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp:201]: Also put info about confidence.

Comment (eizamaliev) [reply demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp:201]: Also align Font, Color, Thikness etc. with OMZ demo.

File: demos/face_detection_mtcnn_demo/cpp_gapi/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+Running the application with an empty list of options yields an error message.
+
+For example, to do inference on a CPU, run the following command:
+
+```sh
+./face_detection_mtcnn_demo_gapi -d_p CPU -i <path_to_video>/input_video.mp4 -m_p mtcnn-p.xml -m_o mtcnn-o.xml -m_r mtcnn-r.xml
+```
+
+>**NOTE**: If you provide a single image as an input, the demo processes and renders it quickly, then exits. To continuously visualize inference results on the screen, apply the `loop` option, which enforces processing a single image in a loop.
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/cpp_gapi/README.md:91]:
```suggestion
>**NOTE**: If you provide a single image as an input, the demo processes and renders it quickly, then exits. To continuously visualize inference results on the screen, apply the `-loop` option, which enforces processing a single image in a loop.
```

File: demos/face_detection_mtcnn_demo/cpp_gapi/README.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  -hs,                  Optional. MTCNN P use half scale pyramid.
+  --loop                Optional. Enable reading the input in a loop.
+  --no_show             Optional. Don't show output
+  -o OUTPUT, --output OUTPUT
+                        Optional. Name of the output file(s) to save.
+  -limit OUTPUT_LIMIT, --output_limit OUTPUT_LIMIT
+                        Optional. Number of frames to store in output. If 0 is
+                        set, all frames are stored.
+  -u UTILIZATION_MONITORS, --utilization_monitors UTILIZATION_MONITORS
+                        Optional. List of monitors to show initially.
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/cpp_gapi/README.md:80]: Please, update.

File: demos/face_detection_mtcnn_demo/cpp_gapi/include/custom_kernels.hpp [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    float score;
+    std::array<float, NUM_REGRESSIONS> regression;
+    std::array<float, 2 * NUM_PTS> ptsCoords;
+
+    static void applyRegression(std::vector<Face>& faces, bool addOne = false) {
+        for (auto& face : faces) {
+            float bboxWidth =
+                face.bbox.x2 - face.bbox.x1 + static_cast<float>(addOne);
+            float bboxHeight =
+                face.bbox.y2 - face.bbox.y1 + static_cast<float>(addOne);
```
Comment (eizamaliev) [inline demos/face_detection_mtcnn_demo/cpp_gapi/include/custom_kernels.hpp:50]:
```suggestion
            float bboxWidth = face.bbox.x2 - face.bbox.x1 + static_cast<float>(addOne);
            float bboxHeight = face.bbox.y2 - face.bbox.y1 + static_cast<float>(addOne);
```

**Review by eizamaliev — DISMISSED** [review]

**Comment by vladimir-dudnik** [comment]
@mpashchenkov I've commit suggestions on your behalf, please update your local branch from remote repo before working on subsequent changes

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp (hunk 2 above)
Comment (mpashchenkov) [reply demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp]: Added.

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp (hunk 3 above)
Comment (mpashchenkov) [reply demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp]: Added unified metrics.

File: demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp (hunk 4 above)
Comment (mpashchenkov) [reply demos/face_detection_mtcnn_demo/cpp_gapi/main.cpp]: Aligned.

File: demos/face_detection_mtcnn_demo/cpp_gapi/README.md (hunk 6 above)
Comment (mpashchenkov) [reply demos/face_detection_mtcnn_demo/cpp_gapi/README.md]: Indentation updated.

Comment (mpashchenkov) [reply demos/face_detection_mtcnn_demo/cpp_gapi/README.md]: Something else?

**Review by vladimir-dudnik — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by mpashchenkov** [comment]
Ok. I can't fix Win10 red build (some problems with Yolo network isn't part of this PR).

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@mpashchenkov yolo warning were fixed in https://github.com/openvinotoolkit/open_model_zoo/pull/2654, please rebase your branch to develop

**Review by vladimir-dudnik — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
cc @Wovchena

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Pull request merged**